src/main/java/teammates/storage/sql/CoursesDb.java

```diff
@@ -139,10 +135,11 @@ CourseAttributes makeAttributes(Course entity) {
     }
 
     private Course getCourseEntity(String courseId) {
-        Session session = HibernateUtil.getSessionFactory().openSession();
-        Course course = session.get(Course.class, courseId);
-        session.close();
-        return course;
+        // Flushing forces the creation date of courses created in the same session
+        // to be automatically updated, otherwise it would be null
+        HibernateUtil.getSessionFactory().getCurrentSession().flush();
```

Comment:
According to this [post](https://stackoverflow.com/a/14322619), it seems that the changes will not be seen after the commit. Therefore, I am not sure whether this is correct. Would like to check whether you encounter this problem during testing and can be resolved by this line?